pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -563,6 +569,55 @@ private BrokerResponseNative handleSQLRequest(long requestId, String query, Json
     return brokerResponse;
   }
 
+  private void handleTimestampIndexOverride(PinotQuery pinotQuery, TableConfig tableConfig) {
```

Comment:
since this happens once per query, I think we should avoid using the stream API here because it's a lot slower than writing imperative code.

Reply:
moved to a util function